modules/ROOT/pages/consume-data-from-an-api.adoc

```diff
@@ -0,0 +1,323 @@
+
+To perform the same operation by using Anypoint Studio, modify the path configuration with the desired DataWeave expression: in this case `#['/users/' ++ payload.user_id]`.
+
+.Configure Dynamic Requests with DataWeave expressions in Path field.
+image::mruntime-consume-rest-api-configuration-3.png[Configure Dynamic Requests with DataWeave expressions in Path field]
+
+You can also configure dynamic requests for the query or URI parameters, or for the HTTP method.
+Additionally, you can use xref:configuring-properties.adoc[Configuration Properties] to set any of these dynamic requests.
+
+=== Configure Dynamic Requests in XML
+
+The following example code in XML, shows the configuration of dynamic requests:
+
+[source,xml,linenums]
+----
+  <http:request-config name="HTTP_Request_configuration">
+    <http:request-connection host="jsonplaceholder.typicode.com" />
+  </http:request-config>
+  <flow name="basic_tutorialFlow1">
+    <...>
+    <set-payload value='#[{user_id: "10"}]'/>
+    <http:request method="GET" config-ref="HTTP_Request_configuration" path="#['/users/' ++ payload.user_id]"/>
+  </flow>
+----
+
+== HTTP Content-Type and Encoding
+
+When you send a POST request, Mule runtime engine adheres to the following rules regarding Content-Type and encoding of the body.
+
+[cols="2*"]
+|===
+|Content |Rules
+|*For a String, char[], Reader, or similar* a|
+* If the endpoint has explicitly set encoding, Mule runtime engine uses that encoding.
+* If the endpoint does not have explicitly set encoding, Mule runtime engine determines the encoding from the message property `Content-Type`.
+* If the `Content-Type` message property is not set, Mule runtime engine uses the Mule Context default configuration.
+* For `Content-Type`, Mule runtime engine sends the message property `Content-Type`, but with the actual encoding set.
+
+|*For binary content* a|
+Encoding is not relevant. Mule runtime engine sets `Content-Type` as follows:
+
+* If the `Content-Type` property is set on the message, Mule runtime engine uses the defined content type.
+* If the `Content-Type` property is not set on the message, Mule runtime engine sets "application/octet-stream" as `Content-Type`.
+
+|===
+
+== Custom Headers
+
+Some APIs require that you pass custom headers, such as your developer key, along with your requests. As with the query parameters, you can also add headers to your request on the HTTP Connector. +
+For example, if the API you are consuming requires a developer key, then specify that key key as a header on your requests by using the header name `accessKey`. +
+You can add a property to set custom headers by using Anypoint Studio.
+
+=== Custom Headers in Anypoint Studio
+
+To set custom headers in Studio, review the following steps:
+
+* In the HTTP Request operation's properties editor, select the *Headers* tab.
+* Click *Add*.
+* Define *Header Name* and its value. Both can be defined using expressions or properties placeholders. +
```

Comment:
Define *Header Name* and its value. > Define *Name* and *Value*
To match what the UI shows, or just remove bold from `Header Name`